### ISSUE TITLE
Some OIDC tests have become flaky

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#27900]
### Describe the bug

Looks like some instability has been introduced after the upgrade to KC 19.0.1 - as no significant changes have been made to `CodeAuthenticationMechanism`.

I've just seen `CodeFlowTest.testTokenAutorefresh` failing in one of the PRs.
I've seen a few more tests failing randomly.

**Comment by sberyozkin** [comment]
`io.quarkus.it.keycloak.CodeFlowTest.testRPInitiatedLogout` is failing as well.

I'll have a look tomorrow, but if necessary a PR updating to KC 19.0.1 can be temp reverted and check if it makes it more stable, @gsmet, let me know please what you prefer

**Comment by gsmet** [comment]
@sberyozkin I plan to disable all the flaky tests on Thursday afternoon. So if you can fix them by then, fine :).

**Comment by sberyozkin** [comment]
@gsmet Sure, I'll investigate, I'm 99% sure it is a test-only problem, might be timeout related used in these tests, so might be able to improve it...

**Comment by gsmet** [comment]
I'm not totally sure it's just that. For some of them, the HTTP status is incorrect.

**Comment by sberyozkin** [comment]
At the moment if the refreshing the token fails then a flow terminating `AuthenticationCompletionException` is thrown leading to `401` - but failing refreshing the token should really re-initiate the login with `302` - I'm changing it in another PR too...

**Comment by sberyozkin** [comment]
`io.quarkus.it.keycloak.CodeFlowTest.testRPInitiatedLogout` fails because Keycloak returns a `"error":"invalid_grant","error_description":"Token is not active"}` message when a refresh token is used to refresh the tokens - in this case `quarkus-oidc` will indeed request a redirect to Keycloak - but the session at the Keycloak level is still active and the redirect loop starts eventually leading to 401.
I think the reason this test passes sometimes is that in some cases the redirect loop is not happening. But neither the refresh is successful. 
I've tweaked the test to verify that the refresh grant was in fact successful and I just can't get the test passing now.

**Comment by sberyozkin** [comment]
I suspect all the other random test failures are related to this inactive refresh token case.
Forgot to add - I managed to get avoid this error once...

**Comment by sberyozkin** [comment]
Keycloak returns a refresh token which is valid for 14 secs, but it fails after about 7 secs

**Comment by sberyozkin** [comment]
`CodeFlowTest` tests have been passing for me today, so I've opened #27968 in an attempt to simplify them a bit, though no guarantee yet they will become stable.

The other test which is randomly failing is `oidc-token-propagation-reactive`: https://github.com/quarkusio/quarkus/pull/27737#issuecomment-1237440403 - I'll have a look next

**Comment by sberyozkin** [comment]
Hi @gsmet I've seen CodeFlowTest#testTokenRefresh (which I added by splitting a complex `#RPInitiatedLogout` into two tests) failing only once since Thursday, so I'd like to keep it running as it does not interfere much, I'd like to add more debug messages today to help tracing what went wrong next time it fails. Some race condition is going on apparently, for now I think it is a awaitility/HTML unit test issue but lets see

**Comment by sberyozkin** [comment]
@gsmet I've seen one of these tests failing the 3rd time since last Thursday in #28109, while the failure frequency is low, a race condition is there and I have to admit even with the added debug logging I've no idea what is causing the failure.
I'm prepared to disable these refresh token tests, just concerned I won't be able to reproduce locally, I haven't so far.

I'd like to add a few more debug messages to `main`, we still have a few days till next week :-)

**Comment by gsmet** [comment]
@sberyozkin no objections on my side.

**Comment by sberyozkin** [comment]
`integration-tests/oidc-token-propagation-reactive` keeps failing, just seeing a strange exception in
```
2022-09-27T14:16:02.1526062Z 2022-09-27 14:16:02,099 INFO  [🐳 .io/.0.2]] (build-26) Container quay.io/keycloak/keycloak:19.0.2 is starting: 54e82cc33cf3df64bf124773c72a89f3fcab06660cbe2a250b00f77662986e4a
2022-09-27T14:16:13.8706157Z 2022-09-27 14:16:13,838 INFO  [🐳 .io/.0.2]] (build-26) Container quay.io/keycloak/keycloak:19.0.2 started in PT11.773018945S
2022-09-27T14:16:14.5714709Z 2022-09-27 14:16:14,536 ERROR [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (build-26) Realm quarkus can not be created: {"error":"invalid_grant","error_description":"Invalid user credentials"}
2022-09-27T14:16:14.5715940Z 2022-09-27 14:16:14,555 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] 
...
```

disabling it now

**Comment by geoand** [comment]
@sberyozkin is this still an issue?

**Comment by geoand** [comment]
Closing
